Guide to use Prometheus collector with Sensu Go (#1058)
sensu/sensu-docs: content/sensu-go/5.0/guides/prometheus-metrics.md

---
title: "Using Prometheus with the Sensu Prometheus Collector"
linkTitle: "Using Prometheus with Sensu"
weight: 120
version: "5.0"
product: "Sensu Go"
platformContent: false
menu: 
  sensu-go-5.0:
    parent: guides
---

- [What is the Sensu Prometheus Collector?](#what-is-the-sensu-prometheus-collector)
- [Install and Configure Prometheus](#install-and-configure-prometheus)
- [Install and Configure Sensu Go](#install-and-configure-sensu-go)
- [Install Sensu Prometheus Collector](#install-sensu-prometheus-collector)
- [Install Sensu InfluxDB Handler](#install-sensu-influxdb-handler)
- [Install and Configure InfluxDB](#install-and-configure-influxdb)
- [Install and Configure Grafana](#install-and-configure-grafana)

## What is the Sensu Prometheus Collector?

> The [Sensu Prometheus Collector][1] is a sensu check plugin that can collect metrics from a [Prometheus exporter][2] or the [Prometheus query API][3]. The allows Sensu to then send metrics to one or more time-series databases, such as InfluxDB or Graphite.

Reviewer: ```suggestion
The [Sensu Prometheus Collector][1] is a check plugin that collects metrics from a [Prometheus exporter][2] or the [Prometheus query API][3]. This allows Sensu to route the collected metrics to one or more time-series databases, such as InfluxDB or Graphite.
```

> 

Reviewer: Feels like we need a section to describe why I would want to do this.

Author: Agreed. As I attempt to write this though, I am finding myself just reiterating what the Prometheus Collector does vs an actual user use case.

Maybe we can pair up on Monday or during office hours to come up with something that is valuable to readers.

## In this guide

This guide uses CentOS 7 as the operating system with all compoments running on the same compute resource. Commands and steps may change for different distributions or if components are running on different compute resources.

> At the end, you will have Prometheus scraping metrics. The Sensu Prometheus Exporter will then query the Prometheus API as a Sensu check, send those to an InfluxDB Sensu handler, which will send metrics to an InfluxDB instance. Finally, Grafana will query InfluxDB to display those collected metrics.

Reviewer: ```suggestion
At the end, you will have Prometheus scraping metrics. The Sensu Prometheus Collector will then query the Prometheus API as a Sensu check, send those to an InfluxDB Sensu handler, which will send metrics to an InfluxDB instance. Finally, Grafana will query InfluxDB to display those collected metrics.
```

tar xvfz /tmp/sensu-influxdb-handler_3.0.1_linux_amd64.tar.gz -C /tmp/
cp /tmp/bin/sensu-influxdb-handler /usr/local/bin/
{{< /highlight >}}

### Install and Configure InfluxDB

Add InfluxDB repo

{{< highlight shell >}}
echo "[influxdb]
name = InfluxDB Repository - RHEL \$releasever
baseurl = https://repos.influxdata.com/rhel/\$releasever/\$basearch/stable
enabled = 1
gpgcheck = 1
gpgkey = https://repos.influxdata.com/influxdb.key" | sudo tee /etc/yum.repos.d/influxdb.repo
{{< /highlight >}}

Install InfluxDB

{{< highlight shell >}}
sudo yum -y install influxdb
{{< /highlight >}}

In `/etc/influxdb/influxdb.conf`, uncomment the http API line

{{< highlight shell >}}
[http]
  # Determines whether HTTP endpoint is enabled.
  enabled = true
{{< /highlight >}}

Start InfluxDB

{{< highlight shell >}}
sudo systemctl start influxdb
{{< /highlight >}}

Add sensu user and sensu DB

{{< highlight shell >}}
influx -execute "CREATE DATABASE sensu"
influx -execute "CREATE USER sensu WITH PASSWORD 'sensu'"
influx -execute "GRANT ALL ON sensu TO sensu"
{{< /highlight >}}

### Add Sensu check and handler to complete the pipeline

> Using the following resource definition in a file called `resources.yml`:

Reviewer: ```suggestion
Given the following resource definition in a file called `resources.yml`:
```

          "values": []
        },
        "yaxes": [
          {
            "format": "short",
            "label": null,
            "logBase": 1,
            "max": null,
            "min": null,
            "show": true
          },
          {
            "format": "short",
            "label": null,
            "logBase": 1,
            "max": null,
            "min": null,
            "show": true
          }
        ],
        "yaxis": {
          "align": false,
          "alignLevel": null
        }
      }
    ],
    "schemaVersion": 16,
    "style": "dark",
    "tags": [],
    "templating": {
      "list": []
    },
    "time": {
      "from": "now-24h",
      "to": "now"
    },
    "timepicker": {
      "refresh_intervals": [
        "5s",
        "10s",
        "30s",
        "1m",
        "5m",
        "15m",
        "30m",
        "1h",
        "2h",
        "1d"
      ],
      "time_options": [
        "5m",
        "15m",
        "1h",
        "6h",
        "12h",
        "24h",
        "2d",
        "7d",
        "30d"
      ]
    },
    "timezone": "",
    "title": "Up or Down Sample 2",
    "uid": "nb442Vsiz",
    "version": 1
  }
}
{{< /highlight >}}

> 

Reviewer: Feels like this needs a conclusion section to wrap things up.

Author: Ah yes, definitely. I'll come up with something.

---
title: "Using Prometheus with the Sensu Prometheus Collector"
linkTitle: "Using Prometheus with Sensu"
weight: 120
version: "5.0"
product: "Sensu Go"
platformContent: false
menu: 
  sensu-go-5.0:
    parent: guides
---

- [What is the Sensu Prometheus Collector?](#what-is-the-sensu-prometheus-collector)
- [Why use Sensu with Prometheus?](#why-use-sensu-with-prometheus)
- [Install and Configure Prometheus](#install-and-configure-prometheus)
- [Install and Configure Sensu Go](#install-and-configure-sensu-go)
- [Install Sensu Prometheus Collector](#install-sensu-prometheus-collector)
- [Install Sensu InfluxDB Handler](#install-sensu-influxdb-handler)
- [Install and Configure InfluxDB](#install-and-configure-influxdb)
- [Install and Configure Grafana](#install-and-configure-grafana)

## What is the Sensu Prometheus Collector?

The [Sensu Prometheus Collector][1] is a check plugin that collects metrics from a [Prometheus exporter][2] or the [Prometheus query API][3]. This allows Sensu to route the collected metrics to one or more time-series databases, such as InfluxDB or Graphite.

## Why use Sensu with Prometheus?

> Prometheus offers an actively maintained [node exportor][5] for reporting hardware and operating system metrics. With those metrics exposed, Sensu can route them to one or more time-series databases, such as InfluxDB or Graphite. Both Sensu and Prometheus can run in parallel, complimenting each other and making use of environments where Prometheus is already deployed.  

Reviewer: ```suggestion
The Prometheus ecosystem contains a number of actively maintained exporters, such as the [node exporter][5] for reporting hardware and operating system metrics or Google's cAdvisor exporter for monitoring containers. These exporters expose metrics which Sensu can collect and route to one or more time-series databases, such as InfluxDB or Graphite. Both Sensu and Prometheus can run in parallel, complimenting each other and making use of environments where Prometheus is already deployed.  
```

Reviewer: Link to https://github.com/google/cadvisor for cAdvisor

At the end, you will have Prometheus scraping metrics. The Sensu Prometheus Collector will then query the Prometheus API as a Sensu check, send those to an InfluxDB Sensu handler, which will send metrics to an InfluxDB instance. Finally, Grafana will query InfluxDB to display those collected metrics.

### Install and Configure Prometheus

Download and extract Prometheus

{{< highlight shell >}}
wget https://github.com/prometheus/prometheus/releases/download/v2.6.0/prometheus-2.6.0.linux-amd64.tar.gz
tar xvfz prometheus-*.tar.gz
cd prometheus-*
{{< /highlight >}}

Setup configuration in a file called `prometheus.yml`

{{< highlight shell >}}
global:
  scrape_interval: 15s
  external_labels:
    monitor: 'codelab-monitor'

scrape_configs:
  - job_name: 'prometheus'
    scrape_interval: 5s
    static_configs:
      - targets: ['localhost:9090']
{{< /highlight >}}

Start Prometheus in the background

{{< highlight shell >}}
nohup .prometheus --config.file=prometheus.yml > prometheus.log 2>&1 &
{{< /highlight >}}

> Ensure Pormetheus is running

Reviewer: ```suggestion
Ensure Prometheus is running
```
